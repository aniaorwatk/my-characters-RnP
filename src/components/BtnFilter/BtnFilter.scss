@import "../../scss/Main";

.btnFilter {
  padding: 5px;
  margin: 5px;
  font-family: Arial, Helvetica, sans-serif;
font-weight: bold;
  color: $color-text-white;
  border-radius: 5px;
  &:hover {
    opacity: 0.9;
  }

}

.colorDnD {
  background-color: $color-DnDdark-purple;

}

.colorCthulhu{
    background-color: $color-Cdark-sea;
}

.colorAll{
    background-color: $color-text-names;
}
.colorDead{
    background-color: $color-text-black;
}
.colorLiving{
    color:$color-text-black;
}
