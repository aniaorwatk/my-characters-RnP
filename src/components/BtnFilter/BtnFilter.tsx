import "./BtnFilter.scss";

interface IBtnFilterType {
  labelBtn: string;
  onClick: () => void;
  btnColor: string;
}
const BtnFilter = ({ labelBtn, btnColor, onClick }: IBtnFilterType) => {
  return (
    <button type="button" className={`btnFilter ${btnColor}`} onClick={onClick}>
      {labelBtn}
    </button>
  );
};

export default BtnFilter;
