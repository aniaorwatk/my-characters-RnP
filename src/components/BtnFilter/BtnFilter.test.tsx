import { render, screen } from '@testing-library/react';
import BtnFilter from './BtnFilter';

test('renders Btn Filter', () => {
    render(<BtnFilter labelBtn='All Characters' onClick={function (): void {
        throw new Error('Function not implemented.');
    }} btnColor='yellow' />);

    const filterButtonElement = screen.getByText(/All Characters/i);

    expect(filterButtonElement).toBeInTheDocument();
});

test('render snapshot', () => {
    render(<BtnFilter labelBtn='All Characters' onClick={function (): void {
        throw new Error('Function not implemented.');
    }} btnColor='yellow' />);

    const filterButtonElement = screen.getByText(/All Characters/i);

    expect(filterButtonElement).toMatchSnapshot();
});

test('when click button function been called', () => {
    const mock = jest.fn();
    render(<BtnFilter labelBtn='All Characters' onClick={mock} btnColor='yellow' />);

    const filterButtonElement = screen.getByText(/All Characters/i);
    filterButtonElement.click();

    expect(mock).toBeCalled();
});