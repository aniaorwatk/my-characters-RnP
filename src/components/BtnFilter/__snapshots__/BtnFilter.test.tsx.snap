// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<button
  class="btnFilter yellow"
  type="button"
>
  All Characters
</button>
`;
